src/Neo/SmartContract/ApplicationEngine.cs

```diff
@@ -613,28 +631,11 @@ public void SetState<T>(T state)
 
         public bool IsHardforkEnabled(Hardfork hardfork)
         {
-            // Return true if there's no specific configuration or PersistingBlock is null
-            if (PersistingBlock is null || ProtocolSettings.Hardforks.Count == 0)
+            // Return true if PersistingBlock is null
+            if (PersistingBlock is null)
                 return true;
```

Comment:
If `C` is not enabled, it must return `false` always.

```suggestion
            // Return true if PersistingBlock is null and Hardfork is enabled
            if (PersistingBlock is null)
                return ProtocolSettings.Hardforks.ContainsKey(hardfork);
```

Reply:
LGTM. I dont think this really matters, as long as it works for the mainnet and testnet, the specific logic about the hardfork configuration can be set in any logic that may align to @AnnaShaleva 's logic or your logic. If you have it set here, then it is the rule.